app/util/appUtil.js

```diff
@@ -140,7 +141,10 @@ async function processMetadata(metadata, files) {
         const keyAsUint8Array = utf8ToUint8Array(key, METADATA_KEY_LENGTH)
 
         const validMetadataValueTypes = Object.keys(apiOptions.types.MetadataValue._enum)
-        if (typeof value !== 'object' || !validMetadataValueTypes.some((type) => type.toUpperCase() === value.type)) {
+        if (
+          typeof value !== 'object' ||
+          !validMetadataValueTypes.some((type) => type.toLowerCase() === value.type.replace(/_/g, '').toLowerCase())
+        ) {
```

Comment:
Couple of comments on this.

Is my understanding of this logic correct and does it do as you'd like? As it will return false if not all match, but there may be some that are ok and then that will skip the error?

Also is line ending necessary between 144 and 145?

Reply:
It's supposed to be checking the `type` of each metadata object to make sure it's within the list of `polkadot.js` types we've set. The first metadata it encounters where its `type` isn't in the list, it throws an error and responds with 400. If some are OK and some aren't, it should 400. But perhaps I'm getting confused and it isn't.

This has made me realise the error message is wrong, needs to add _ like `TokenId`->`TOKEN_ID`. Maybe it would be better to just use CamelCase for `type` in requests to match the node.

By line ending 144, do you mean does 145 need to start on a new line? That's coming from prettier

Reply:
So my understanding is:

* `value` may be a string or an object. If it is an object it must have a property `type` which exists in the list of valid metadata types. If `value` is a string it refers to a metadata file name in the multipart body
* the break between 144 and 145 is driven by prettier. It is what it is

There is an issue here though. What happens if `value` is `null`?

Reply:
One question though is why the shift to lowercasing on both sides? I'd generally prefer the uppercasing on the left only here as it was before

Reply:
> One question though is why the shift to lowercasing on both sides? I'd generally prefer the uppercasing on the left only here as it was before

yep not sure why I've done that, will put uppercasing back. If we're sticking with uppercase shall I remove the map on L149. Converting `TokenId` -> `TOKEN_ID` seems to require some complex regex

Reply:
There's some potential confusion over what `value` means here. Value within this function block is the `value` from a metadata key:value pair e.g. if metadata is
 
```
"metadata": {
  "some_file": { "type": "FILE", "value": "some_file.txt"},
}
```

value here means `{ "type": "FILE", "value": "some_file.txt"}` not `value` within the object

Reply:
https://github.com/digicatapult/vitalam-api/pull/16/commits/33357d8cf80c77e4b6a2da4ba44d9540f56cde4f

^ result of talking through all these comments with Matt